wormhole/file_transport.go

```diff
@@ -251,30 +261,33 @@ func (t *fileTransport) connectDirect(otherTransit *transitMsg) (net.Conn, error
 	return conn, nil
 }
 
-func (t *fileTransport) connectToRelay(ctx context.Context, successChan chan net.Conn, failChan chan string) {
+func (t *fileTransport) connectToRelay(ctx context.Context, addr string, successChan chan net.Conn, failChan chan string) {
 	var d net.Dialer
 	var conn net.Conn
 	var err error
-	addr := t.relayURL.Host // This contains the port if necessary
+
+	//in case address is not provide in hints
+	if addr == "" {
+		addr = t.relayURL.Host
+	}
 
 	switch t.relayURL.Scheme {
 	case "tcp":
 		conn, err = d.DialContext(ctx, "tcp", addr)
-
 		if err != nil {
 			failChan <- addr
 			return
 		}
+		fmt.Println("Downloading... via TCP relay")
 	case "ws", "wss":
 		var wsconn *websocket.Conn
-		wsconn, _, err = websocket.Dial(ctx, t.relayURL.String(), nil)
-
+		wsconn, _, err = websocket.Dial(ctx, addr, nil)
```

Comment:
You need to pass a full URL to `websocket.Dial`: the URL may contain a path and query parameters, so only the host address won't be enough.

Reply:
switched to URL, should be ok now.